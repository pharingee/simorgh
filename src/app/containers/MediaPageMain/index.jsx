import React, { Fragment, useContext } from 'react';
import { string, shape, object } from 'prop-types';
import { Headline } from '@bbc/psammead-headings';
import Paragraph from '@bbc/psammead-paragraph';
import pathOr from 'ramda/src/pathOr';

import MetadataContainer from '../Metadata';
import { Grid, GridItemConstrainedMedium } from '../../lib/styledGrid';
import { ServiceContext } from '../../contexts/ServiceContext';

const MediaPageMain = props => {
  const { pageData, service, match } = props;
  const { serviceId, mediaId } = match.params;
  const { script, translations } = useContext(ServiceContext);

  const { title, subtitle } = pathOr(null, ['media', serviceId], translations);

  return (
    <Fragment>
      <MetadataContainer metadata={pageData.metadata} promo={pageData.promo} />
      <main role="main">
        <Grid>
          <GridItemConstrainedMedium>
            <Headline script={script} service={service}>
              {title}
            </Headline>
            <Paragraph script={script} service={service}>
              {subtitle}
            </Paragraph>
            <ul>
              <li>
                <strong>Service</strong>: {service}
              </li>
              <li>
                <strong>Brand</strong>: {serviceId}
              </li>
              <li>
                <strong>MediaId</strong>: {mediaId}
              </li>
            </ul>
          </GridItemConstrainedMedium>
        </Grid>
      </main>
    </Fragment>
  );
};

MediaPageMain.propTypes = {
  service: string.isRequired,
  pageData: shape({
    metadata: shape({
      id: string,
      tags: object,
    }),
    promo: shape({
      subtype: string,
      name: string,
    }),
  }).isRequired,
  match: shape({
    params: shape({
      serviceId: string,
      mediaId: string,
    }),
  }).isRequired,
};

export default MediaPageMain;
