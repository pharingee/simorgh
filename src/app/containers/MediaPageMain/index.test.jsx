import React from 'react';
import MediaPageMain from '.';
import { shouldMatchSnapshot } from '../../../testHelpers';
import amharicConfig from '../../lib/config/services/amharic';

const liveRadioScaffoldProps = {
  service: 'amharic',
  pageData: {
    metadata: {
      id: 'bbc_amharic_radio',
      tags: {},
    },
    promo: {
      subType: 'IDX',
      name: 'BBC Amharic Radio',
      uri: '/amharic/bbc_amharic_radio/liveradio',
      id: 'some-id',
      type: 'simple',
    },
  },
  match: {
    params: {
      serviceId: 'bbc_amharic_radio',
      mediaId: 'liveradio',
    },
  },
};

jest.mock('react', () => {
  const original = jest.requireActual('react');
  return {
    ...original,
    useContext: jest.fn(),
  };
});

const { useContext } = jest.requireMock('react');
useContext.mockReturnValue(amharicConfig);

describe('Media Page Main', () => {
  describe('snapshots', () => {
    shouldMatchSnapshot(
      'should match scaffold snapshot',
      <MediaPageMain {...liveRadioScaffoldProps} />,
    );
  });
});
