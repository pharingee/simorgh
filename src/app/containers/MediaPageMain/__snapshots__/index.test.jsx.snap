// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Page Main snapshots should match scaffold snapshot 1`] = `
.c2 {
  font-size: 1.75rem;
  line-height: 2rem;
  color: #3F3F42;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  margin: 0;
  padding: 2rem 0;
  font-weight: 500;
}

.c3 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
}

.c0 {
  margin: 0 auto;
  padding-bottom: 2rem;
}

.c1 {
  grid-column: 1 / span 6;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 2rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 2.75rem;
    line-height: 3rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    padding: 2.5rem 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (max-width:37.4375rem) {
  .c0 {
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column-gap: 1rem;
  }
}

@media (max-width:62.9375rem) {
  .c0 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(8,minmax(0,6.75rem)) 1fr;
    max-width: 45.5rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(20,minmax(0,2.95rem)) 1fr;
    max-width: 46.4rem;
  }
}

@supports (display:grid) {
  .c0 {
    display: grid;
    max-width: initial;
    margin: initial;
  }
}

@media (max-width:25rem) {
  .c1 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c1 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c1 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c1 {
    max-width: initial;
  }
}

<main
  role="main"
>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <h1
        className="c2"
      >
        ያድምጡ
      </h1>
      <p
        className="c3"
      >
        ዝግጅቶቻችንን
      </p>
      <ul>
        <li>
          <strong>
            Service
          </strong>
          : 
          amharic
        </li>
        <li>
          <strong>
            Brand
          </strong>
          : 
          bbc_amharic_radio
        </li>
        <li>
          <strong>
            MediaId
          </strong>
          : 
          liveradio
        </li>
      </ul>
    </div>
  </div>
</main>
`;
